externs/browser/streamsapi.js

```diff
@@ -14,294 +14,446 @@
- * @return {!ReadableStream}
+ * The ReadableStreamBYOBReader constructor is generally not meant to be used directly;
+ * instead, a stream’s getReader() method should be used.
+ * 
+ * @interface
+ * @see https://streams.spec.whatwg.org/#byob-reader-class
  */
-ReadableStream.prototype.pipeThrough = function(transform, opt_options) {};
+function ReadableStreamBYOBReader() {};
 
 /**
- * @param {!WritableStream} dest
- * @param {PipeOptions=} opt_options
+ * @type {!Promise<void>}
+ * @see https://streams.spec.whatwg.org/#byob-reader-closed
+ */
+ReadableStreamBYOBReader.prototype.closed;
+
+/**
+ * @param {*} reason
  * @return {!Promise<*>}
+ * @see https://streams.spec.whatwg.org/#byob-reader-cancel
  */
-ReadableStream.prototype.pipeTo = function(dest, opt_options) {};
+ReadableStreamBYOBReader.prototype.cancel = function(reason) {};
 
 /**
- * @return {!Array<!ReadableStream>} Two-element array, containing resulting
- *     streams.
+ * @param {!ArrayBufferView} view
+ * @return {!Promise<!IteratorResult>}
+ * @see https://streams.spec.whatwg.org/#byob-reader-read
  */
-ReadableStream.prototype.tee = function() {};
+ReadableStreamBYOBReader.prototype.read = function(view) {};
 
 /**
- * @see https://streams.spec.whatwg.org/#rs-controller-class
- * @param {!ReadableStream} stream
- * @constructor
+ * @return {undefined}
+ * @see https://streams.spec.whatwg.org/#byob-reader-release-lock
  */
-function ReadableStreamController(stream) {}
+ReadableStreamBYOBReader.prototype.releaseLock = function() {};
 
-/** @type {number} */
-ReadableStreamController.prototype.desiredSize;
 
-/** @return {undefined} */
-ReadableStreamController.prototype.close = function() {};
+/**
+ * The ReadableStreamDefaultController constructor cannot be used directly;
+ * it only works on a ReadableStream that is in the middle of being constructed.
+ * 
+ * @interface
+ * @see https://streams.spec.whatwg.org/#rs-default-controller-class
+ */
+function ReadableStreamDefaultController() {};
+
+/**
+ * @type {number}
+ * @see https://streams.spec.whatwg.org/#rs-default-controller-desired-size
+ */
+ReadableStreamDefaultController.prototype.desiredSize;
 
 /**
- * @param {*} chunk
  * @return {undefined}
+ * @see https://streams.spec.whatwg.org/#rs-default-controller-close
  */
-ReadableStreamController.prototype.enqueue = function(chunk) {};
+ReadableStreamDefaultController.prototype.close = function() {};
 
 /**
- * @param {*} e
+ * @param {*} chunk
  * @return {undefined}
+ * @see https://streams.spec.whatwg.org/#rs-default-controller-enqueue
  */
-ReadableStreamController.prototype.error = function(e) {};
+ReadableStreamDefaultController.prototype.enqueue = function(chunk) {};
 
 /**
- * @see https://streams.spec.whatwg.org/#reader-class
- * @param {!ReadableStream} stream
- * @constructor
+ * @param {*} err
+ * @return {undefined}
+ * @see https://streams.spec.whatwg.org/#rs-default-controller-error
  */
-function ReadableStreamReader(stream) {}
```

Comment:
There are a couple of projects referencing this type by name so we need to change them to `ReadableStreamDefaultReader` which is why this is taking a little bit of time to submit internally. Sorry for the delay.